x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/common/Failure.java

```diff
@@ -51,6 +58,16 @@ public boolean equals(Object obj) {
         }
 
         Failure other = (Failure) obj;
+
+        // When deduplicating failures, it's important to ignore the NameIds of UnresolvedAttributes.
+        // Otherwise, two failures will be emitted to the user for e.g.
+        // `FROM test | STATS max(unknown) by unknown` because the two `unknown` attributes will have differentNameIds - even though they
+        // clearly refer to the same problem.
+        if (node instanceof UnresolvedAttribute ua
+            && other.node instanceof UnresolvedAttribute otherUa
+            && ua.getClass() == otherUa.getClass()) {
+            return ua.nonSemanticEquals(otherUa);
+        }
```

Comment:
This is not a boilerplate change. The old behavior is intentionally kept intact for the Failure class (otherwise, we genuinely fail a test).